Bitbucket.Authentication/BasicAuth/BasicAuthAuthenticator.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Atlassian.Bitbucket.Authentication.Rest;
+using Microsoft.Alm.Authentication;
+using Microsoft.Alm.Git;
+
+namespace Atlassian.Bitbucket.Authentication.BasicAuth
+{
+    class BasicAuthAuthenticator
+    {
+        public async Task<AuthenticationResult> GetAuthAsync(TargetUri targetUri, TokenScope scope, int requestTimeout, Uri restRootUrl, string username, string password)
+        {
```

Comment:
New class that contains the basic Auth code. Mirrors the structure of the OAuthAuthenticator for consistency